### PR TITLE
Dev brand extension

### PR DESCRIPTION
**Pull request by Tianhao-Gu**
Screenshot 2024-07-24 at 6 23 20 PM

### CODE REVIEW

**Comment by Tianhao-Gu** [comment]
* The original readme has step by step instruction of how to build the extension. 
* It also has a link to a [blog](https://blog.ouseful.info/2022/04/29/custom-branded-logos-for-jupyterlab-and-retrolab-jupyter-notebook-v-7/) to explain what files need to be updated. 
* I exchanged every `jupyterlab_ou_brand_extension` to `cdm_jupyterlab_brand_extension` in the repo. 
* Then removed builds, images, icons from previous repo. Added kbase images/icons
* After running build instructions, a new build (cdm_jupyterlab_brand_extension-0.1.0-py3-none-any.whl) is created. And this will be the package we will use for our jupyterlab.

File: .binder/.github/workflows/build.yml
Comment (Tianhao-Gu) [inline .binder/.github/workflows/build.yml:1]: No GHA required for our build - remove dir (I am actually not sure what does .blinder folder do)

File: .binder/.github/workflows/jupyterlite-pages.yml
Comment (Tianhao-Gu) [inline .binder/.github/workflows/jupyterlite-pages.yml:1]: No GHA required for our build - remove dir

File: .binder/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 jupyterlab
 
-./dist/jupyterlab_ou_brand_extension-0.1.0-py3-none-any.whl
\ No newline at end of file
+./dist/cdm_jupyterlab_brand_extension-0.1.0-py3-none-any.whl
```
Comment (Tianhao-Gu) [inline .binder/requirements.txt:3]: This is the newly built package and will be the package we install with our jupyterlab.

File: .github/workflows/build.yml
Comment (Tianhao-Gu) [inline .github/workflows/build.yml:1]: No GHA required for our build - remove dir

File: .github/workflows/jupyterlite-pages.yml
Comment (Tianhao-Gu) [inline .github/workflows/jupyterlite-pages.yml:1]: No GHA required for our build - remove dir

File: MANIFEST.in
Comment (Tianhao-Gu) [inline MANIFEST.in:1]: Replace all occurrences of `jupyterlab_ou_brand_extension` with `cdm_jupyterlab_brand_extension`

File: README.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ Demo [RetroLab / notebook v.7 site via JupyterLite](http://innovationoutside.git
 
 To install the prebuilt extension from a wheel:
 
-`pip3 install --upgrade https://raw.githubusercontent.com/innovationOUtside/jupyterlab_ou_brand_extension/main/dist/jupyterlab_ou_brand_extension-0.1.0-py3-none-any.whl`
+`pip3 install --upgrade https://raw.githubusercontent.com/kbaseincubator/cdm_jupyterlab_brand_extension/blob/main/dist/cdm_jupyterlab_brand_extension-0.1.0-py3-none-any.whl`
```
Comment (Tianhao-Gu) [inline README.md:10]: once PR is merged, jupyterlab repo can install the package from `kbaseincubator`

File: binder/environment.yml
Comment (Tianhao-Gu) [inline binder/environment.yml:1]: Replace all occurrences of `jupyterlab_ou_brand_extension` with `cdm_jupyterlab_brand_extension`

File: binder/postBuild
Comment (Tianhao-Gu) [inline binder/postBuild:1]: Replace all occurrences of `jupyterlab_ou_brand_extension` with `cdm_jupyterlab_brand_extension`

File: dist/cdm_jupyterlab_brand_extension-0.1.0-py3-none-any.whl
Comment (Tianhao-Gu) [inline dist/cdm_jupyterlab_brand_extension-0.1.0-py3-none-any.whl:1]: This is the newly built package.

File: dist/jupyterlab_ou_brand_extension-0.1.0-py3-none-any.whl
Comment (Tianhao-Gu) [inline dist/jupyterlab_ou_brand_extension-0.1.0-py3-none-any.whl:1]: remove old pre-built package with old logos/icons.

File: dist/jupyterlab_ou_brand_extension-0.1.0.tar.gz
Comment (Tianhao-Gu) [inline dist/jupyterlab_ou_brand_extension-0.1.0.tar.gz:1]: remove old pre-built package with old logos/icons.

File: dist/jupyterlab_ou_brand_extension-0.2.0-py3-none-any.whl
Comment (Tianhao-Gu) [inline dist/jupyterlab_ou_brand_extension-0.2.0-py3-none-any.whl:1]: remove old pre-built package with old logos/icons.

File: dist/jupyterlab_ou_brand_extension-0.2.0.tar.gz
Comment (Tianhao-Gu) [inline dist/jupyterlab_ou_brand_extension-0.2.0.tar.gz:1]: remove old pre-built package with old logos/icons.

File: dist/jupyterlab_ou_brand_extension-0.2.1-py3-none-any.whl
Comment (Tianhao-Gu) [inline dist/jupyterlab_ou_brand_extension-0.2.1-py3-none-any.whl:1]: remove old pre-built package with old logos/icons.

File: dist/jupyterlab_ou_brand_extension-0.2.1.tar.gz
Comment (Tianhao-Gu) [inline dist/jupyterlab_ou_brand_extension-0.2.1.tar.gz:1]: remove old pre-built package with old logos/icons.

File: dist/jupyterlab_ou_brand_extension-0.2.2-py3-none-any.whl
Comment (Tianhao-Gu) [inline dist/jupyterlab_ou_brand_extension-0.2.2-py3-none-any.whl:1]: remove old pre-built package with old logos/icons.

File: dist/jupyterlab_ou_brand_extension-0.2.2.tar.gz
Comment (Tianhao-Gu) [inline dist/jupyterlab_ou_brand_extension-0.2.2.tar.gz:1]: remove old pre-built package with old logos/icons.

File: dist/jupyterlab_ou_brand_extension-0.2.3-py3-none-any.whl
Comment (Tianhao-Gu) [inline dist/jupyterlab_ou_brand_extension-0.2.3-py3-none-any.whl:1]: remove old pre-built package with old logos/icons.

File: install.json
Comment (Tianhao-Gu) [inline install.json:1]: Replace all occurrences of `jupyterlab_ou_brand_extension` with `cdm_jupyterlab_brand_extension`

File: jupyterlab_ou_brand_extension/__init__.py
Comment (Tianhao-Gu) [inline jupyterlab_ou_brand_extension/__init__.py:1]: remove the old unrelated dir

File: jupyterlab_ou_brand_extension/_version.py
Comment (Tianhao-Gu) [inline jupyterlab_ou_brand_extension/_version.py:1]: remove the old unrelated dir

File: lib/index.js
Comment (Tianhao-Gu) [inline lib/index.js:1]: Replace all occurrences of `jupyterlab_ou_brand_extension` with `cdm_jupyterlab_brand_extension`

File: package.json
Comment (Tianhao-Gu) [inline package.json:1]: Replace all occurrences of `jupyterlab_ou_brand_extension` with `cdm_jupyterlab_brand_extension`

File: requirements-jupyterlite.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 jupyterlab~=3.3.0
 jupyterlite==0.1.0b4
 
-./dist/jupyterlab_ou_brand_extension-0.1.0-py3-none-any.whl
+./dist/cdm_jupyterlab_brand_extension-0.1.0-py3-none-any.whl
```
Comment (Tianhao-Gu) [inline requirements-jupyterlite.txt:4]: replace old package name to our newly built package name

File: setup.py
Comment (Tianhao-Gu) [inline setup.py:1]: Replace all occurrences of `jupyterlab_ou_brand_extension` with `cdm_jupyterlab_brand_extension`

File: style/base.css
Comment (Tianhao-Gu) [inline style/base.css:1]: update image/icon path to use the KBase image/icon

File: style/images/OU-logo-36x28-old.png
Comment (Tianhao-Gu) [inline style/images/OU-logo-36x28-old.png:1]: remove old unrelated image/logos

File: style/images/OU-logo-36x28.png
Comment (Tianhao-Gu) [inline style/images/OU-logo-36x28.png:1]: remove old unrelated image/logos

File: style/images/OU-logo-43x51-old.png
Comment (Tianhao-Gu) [inline style/images/OU-logo-43x51-old.png:1]: remove old unrelated image/logos

File: style/images/OU-logo-43x51.png
Comment (Tianhao-Gu) [inline style/images/OU-logo-43x51.png:1]: remove old unrelated image/logos

File: style/images/OU-logo-53x42-old.png
Comment (Tianhao-Gu) [inline style/images/OU-logo-53x42-old.png:1]: remove old unrelated image/logos

File: style/images/OU-logo-53x42.png
Comment (Tianhao-Gu) [inline style/images/OU-logo-53x42.png:1]: remove old unrelated image/logos

File: style/images/OU-logo-83x65.png
Comment (Tianhao-Gu) [inline style/images/OU-logo-83x65.png:1]: remove old unrelated image/logos

File: style/images/kbase_logo.png
Comment (Tianhao-Gu) [inline style/images/kbase_logo.png:1]: KBase logo will be used in the built

File: style/index.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ let link = document.createElement('link')
 let oldLink = document.getElementsByClassName('favicon');
 link.rel = 'icon';
 link.type = 'image/x-icon';
-link.href = 'https://raw.githubusercontent.com/innovationOUtside/jupyterlab_ou_brand_extension/main/style/images/favicon.ico';
+link.href = 'https://www.kbase.us/wp-content/uploads/sites/6/2020/08/kbase-favicon-32-bgwhite_rounded_corners.png';
```
Comment (Tianhao-Gu) [inline style/index.js:11]: Borrowed from the KBase Narrative repo.
https://github.com/kbase/ui/blob/main/public/index.html#L14

File: yarn.lock
Comment (Tianhao-Gu) [inline yarn.lock:1]: Updated automatically after I ran the build steps

File: README.md [hunk 5]
```diff
@@ -20,6 +20,21 @@ Custom logos applied to:
 
 About: [Custom Branded Logos for JupyterLab and RetroLab (Jupyter notebook v.7)](https://blog.ouseful.info/2022/04/29/custom-branded-logos-for-jupyterlab-and-retrolab-jupyter-notebook-v-7/)
 
+## Customizing the Extension
```
Comment (Tianhao-Gu) [inline README.md:23]: @MrCreosote Added instructions for customizing the extension.

**Comment by MrCreosote** [comment]
I don't think I'll be able to understand this PR without going through the instructions and doing the whole thing from scratch. As such, I just have one question: given what's in the readme, could a non-js dev like myself, without help, easily replicate what you did here?

**Comment by Tianhao-Gu** [comment]
> I don't think I'll be able to understand this PR without going through the instructions and doing the whole thing from scratch. As such, I just have one question: given what's in the readme, could a non-js dev like myself, without help, easily replicate what you did here?

There will definitely be a learning curve. I didn't know anything about JavaScript or Node.js either, but I followed the original README and got to this point.

**Comment by MrCreosote** [comment]
> There will definitely be a learning curve.

Ok, is there any more information you can add to the readme to reduce the curve?

**Comment by Tianhao-Gu** [comment]
> > There will definitely be a learning curve.
> 
> Ok, is there any more information you can add to the readme to reduce the curve?

There likely are, but I don't think we need to include every detail starting from installing Node.js. If someone cannot follow the original README, I don't think that person should touch this repo.

**Review by MrCreosote — COMMENTED** [review]
Approving mostly because I feel like we've spent too much time going back and forth on this but I'm still trepidatious

**Review by MrCreosote — APPROVED** [review]
urg

**Pull request merged**